scripts/updateapilist.py

```diff
@@ -104,52 +137,45 @@ def generate_table_contents(clients: List[CloudClient]) -> List[str]:
     return content_rows + pypi_links
 
 
-REPO_METADATA_URL_FORMAT = (
-    "https://raw.githubusercontent.com/{repo_slug}/main/.repo-metadata.json"
-)
+def allowed_repo(repo) -> bool:
+    return (
+        repo[REPO_RESPONSE_KEY].startswith("googleapis/python-")
+        and repo[REPO_RESPONSE_KEY] not in REPO_EXCLUSION
+        and not repo[REPO_RESPONSE_KEY]
```

Comment:
And also, the third clause in the `and` means the value has to be `None`, so that means the second clause will always be true. What am I missing?

Reply:
@vchudnov-g 
That's a good catch! The last statement should exclude archived repos.

Essentially, this statement checks that the split repo has a name which starts with `python-`, is not one of the excluded repos that we've defined, and is not archived.

Reply:
My question still stands:

The third clause in the `and` means that for the `and` to evaluate to `True`, the value of `repo[REPO_RESPONSE_KEY]` has to be `None` (or `False`), so that means the second clause would  be true, and the first clause would cause an error.

Or, said another way, if `repo[REPO_RESPONSE_KEY]` is truthy, the third clause will make the whole `and` be `False` and the repo will be returned as disallowed.

It seems to me you probably mean something like this for the `and` part (see my other suggestion about including the mono-repo)

```python
        repo[REPO_RESPONSE_KEY]
        and repo[REPO_RESPONSE_KEY].startswith("googleapis/python-")
        and repo[REPO_RESPONSE_KEY] not in REPO_EXCLUSION
        and not repo[REPO_RESPONSE_KEY].archived # I don't know the semantics of this; you get my intent
```

That said, the generated `README` in this PR includes split repos, and presumably by calling this function, so I'm very puzzled how this ever returns `True`.

Reply:
Good catch! This seems like a copy paste error.

before it was
```
        repo["full_name"].startswith("googleapis/python-")
        and repo["full_name"] not in REPO_EXCLUSION
        and not repo["archived"]
```

but now we have the following  where `REPO_RESPONSE_KEY = "full_name"`
```
        repo[REPO_RESPONSE_KEY].startswith("googleapis/python-")
        and repo[REPO_RESPONSE_KEY] not in REPO_EXCLUSION
        and not repo[REPO_RESPONSE_KEY]
```

but it should be 
```
        repo[REPO_RESPONSE_KEY].startswith("googleapis/python-")
        and repo[REPO_RESPONSE_KEY] not in REPO_EXCLUSION
        and not repo[ARCHIVED_RESPONSE_KEY]
```

where `ARCHIVED_RESPONSE_KEY = "archived"`

Reply:
This was somehow missed in the previous commit. Addressed now!